stl/inc/algorithm

```diff
@@ -10104,8 +10104,8 @@ namespace ranges {
                     _STD move(_UFirst), _STD move(_ULast), _Pass_fn(_Pred), _Pass_fn(_Proj));
                 // Avoid repeatedly initializing objects from the result of an iterator dereference when doing so might
                 // not be idempotent. The `if constexpr` to avoid the extra branch in cases where it's not needed
-                if constexpr (same_as<remove_cvref_t<range_reference_t<_Rng>>, _Vty>
-                              && !is_lvalue_reference_v<range_reference_t<_Rng>>) {
+                if constexpr (!same_as<remove_cvref_t<range_reference_t<_Rng>>, _Vty>
+                              || !is_lvalue_reference_v<range_reference_t<_Rng>>) {

```

Comment:
We don't need to guard prvalue iterators, only xvalue iterators, so `Iis_lvalue_reference_v` here should instead be `is_rvalue_reference_v`:
```suggestion
                              || is_rvalue_reference_v<range_reference_t<_Rng>>) {
```